.github/workflows/codeql-analysis.yml

```
@@ -9,6 +9,11 @@ on:
   schedule:
     - cron: '0 13 * * 1'
 
+permissions:
+  contents: read
+  security-events: write
+  pull-requests: read
```

```suggestion
  security-events: write
```

It seems to that this is enough